WordPress/src/main/java/org/wordpress/android/ui/stats/StatsAbstractFragment.java

```diff
@@ -121,18 +143,123 @@ public void onCreate(Bundle savedInstanceState) {
     }
 
     @Override
+    public void onAttach(Activity activity) {
+        super.onAttach(activity);
+        mResourceVars = new StatsResourceVars(activity);
+    }
+
+    @Override
     public void onSaveInstanceState(Bundle outState) {
        /* AppLog.d(AppLog.T.STATS, this.getClass().getCanonicalName() + " > saving instance state");
         AppLog.d(AppLog.T.STATS, "mStatsTimeframe: " + mStatsTimeframe.getLabel());
         AppLog.d(AppLog.T.STATS, "mDate: " + mDate); */
+
         outState.putString(ARGS_SELECTED_DATE, mDate);
         outState.putSerializable(ARGS_TIMEFRAME, mStatsTimeframe);
+        saveStatsData(outState); // Each fragment will override this
         super.onSaveInstanceState(outState);
     }
 
+    @Override
+    public void onResume() {
+        super.onResume();
+
+        // Init the UI
+        if (hasDataAvailable()) {
+            updateUI();
+        } else {
+            showPlaceholderUI();
+            refreshStats();
+        }
+    }
+
+    @Override
+    public void onStart() {
+        super.onStart();
+        EventBus.getDefault().register(this);
+    }
+
+    @Override
+    public void onStop() {
+        EventBus.getDefault().unregister(this);
+        super.onStop();
+    }
+
+    public boolean shouldUpdateFragmentOnUpdateEvent(StatsEvents.SectionUpdatedAbstract event) {
+        if (!isAdded()) {
+            return false;
+        }
+
+        if (!getDate().equals(event.mDate)) {
+            return false;
+        }
+
+        if (!isSameBlog(event)) {
+            return false;
+        }
+
+        if (event.mTimeframe != getTimeframe()) {
```

Comment:
I changed it to `equals` even thought `==` seems safer and faster. Details here: http://stackoverflow.com/a/2937561